source/geh_calculated_measurements/src/geh_calculated_measurements/testing/utilities/job_tester.py

```
@@ -0,0 +1,146 @@
+        self.config = environment_configuration
+        self.ws = WorkspaceClient(host=self.config.workspace_url, token=self.config.databricks_token)
+        self.job_name = job_name
+        self.job_parameters = job_parameters
+
+        # Configure Azure resources
+        credentials = DefaultAzureCredential()
+        secret_client = SecretClient(
+            vault_url=f"https://{self.config.shared_keyvault_name}.vault.azure.net/",
+            credential=credentials,
+        )
+        self.azure_logs_query_client = LogsQueryClient(credentials)
+        azure_log_analytics_workspace_id = secret_client.get_secret("log-shared-workspace-id").value
+        assert azure_log_analytics_workspace_id is not None, "The Azure log analytics workspace ID cannot be empty."
+        self.azure_log_analytics_workspace_id = azure_log_analytics_workspace_id
+
+    def _get_job_by_name(self, job_name: str) -> BaseJob:
+        jobs = list(self.ws.jobs.list(name=job_name))
+        if len(jobs) == 0:
+            raise ValueError(f"No job found with name {job_name}.")
+        if len(jobs) > 1:
+            raise ValueError(f"Multiple jobs found with name {job_name}.")
+        return jobs[0]
+
+    def start_job(self) -> Wait[Run]:
+        base_job = self._get_job_by_name(self.job_name)
+        params = [f"--{key}={value}" for key, value in self.job_parameters.items()]
+        job = self.ws.jobs.run_now(job_id=base_job.job_id, python_params=params)
+        return job
+
+    def wait_for_job_completion(self, job: Wait[Run], timeout: int = 15) -> RunResultState | None:
+        run = job.result(timeout=timedelta(minutes=timeout))
+        return run.state.result_state
+
+    def run_job_and_wait(self) -> Run:
+        base_job = self._get_job_by_name(self.job_name)
+        params = [f"--{key}={value}" for key, value in self.job_parameters.items()]
+        return self.ws.jobs.run_now_and_wait(job_id=base_job.job_id, python_params=params)
+
+    def execute_statement(
```

what is reason for not using the implementation from geh_common?

---

(DatabricksApiClient)

---

Mostly because the start_job method swallowed the error, which made me doubt the implementation as whole.

---

could it then make sense to move this to geh_common? I know that Volt also use execute_statement.